From 46cd21f9cf24774f28c3eb15e0755b9a200e532a Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>
Date: Thu, 8 Oct 2015 08:38:44 -0700
Subject: Fix LP: #1483410

 This patch corrects a simple bug in the test suite loader that was previously
 silently compesated by eariler versions of the standard library. The bug has
 caused us to effectively import modules and pacakges from _inside_ the
 checkbox_support package. For example, the module 'checkbox_support.dbus'
 was imported simply as 'dbus'. This caused all kinds of understandable havoc.
 The fix relies on the second, optional argument to defaultTestLoader.discover()
 which lets us specify that we want to load 'checkbox_support' from the parent
 directory.
Origin: upstream
Bug-Ubuntu: https://launchpad.net/bugs/1483410
Forwarded: no
Last-Update: 2015-08011

Patch-Name: fix-lp1483410-part1
---
 checkbox_support/tests/__init__.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/checkbox_support/tests/__init__.py b/checkbox_support/tests/__init__.py
index fe0bf3d..2ade626 100644
--- a/checkbox_support/tests/__init__.py
+++ b/checkbox_support/tests/__init__.py
@@ -40,8 +40,9 @@ def load_unit_tests():
     """
     # Discover all unit tests. By simple convention those are kept in
     # python modules that start with the word 'test_' .
-    return defaultTestLoader.discover(
-        os.path.dirname(getabsfile(checkbox_support)))
+    start_dir = os.path.dirname(getabsfile(checkbox_support))
+    top_level_dir = os.path.normpath(os.path.join(start_dir, '..'))
+    return defaultTestLoader.discover(start_dir, top_level_dir=top_level_dir)
 
 
 def test_suite():
